app/controllers/settings/external_cert_controller.rb

```diff
@@ -355,10 +357,56 @@ def warning_subjectaltname(cert, warning_messages, required_san_array, service_n
   end
   # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
 
-  # # Placeholder for trust chain validation
-  # def trust_chain_verify(_cert)
-  #   true
-  # end
+  def trust_chain_verify(cert)
+    _trust_chain_verify(cert, _build_system_wide_certs_store)
+  end
+
+  def _build_system_wide_certs_store
+    cert_store = OpenSSL::X509::Store.new
+
+    system_certs = SystemCertificate.all
+    system_certs.each do |system_cert|
+      cert = system_cert.certificate
+      cert_text = cert.certificate
+      cert_obj = read_cert(cert_text)
+      # Certs must pass parsing to get into the SystemCertificate table; so we ignore failing parse
+      next unless cert_obj
+      cert_store.add_cert(cert_obj)
+    end
+
+    cert_store
+  end
+
+  # rubocop:disable Lint/RescueException
+  def _trust_chain_verify(cert_obj, cert_store)
+    verify_error = ""
+
+    # setup callback
+    cert_store.verify_callback = proc do |preverify_ok, ssl_context|
+      begin
+        if preverify_ok != true || ssl_context.error != 0
+          cert_being_checked = ssl_context.chain[ssl_context.error_depth]
+          failed_cert_subject = cert_being_checked.subject
+          err_msg = "SSL Verification failed: #{ssl_context.error_string}"\
+                    " (#{ssl_context.error}) while verifying #{failed_cert_subject}"
+          verify_error += err_msg
+          false
+        else
+          true
+        end
+      rescue Exception
+        verify_error += err_msg
+        false
+      end
+    end
+
+    if cert_store.verify(cert_obj) == true
```

Comment:
Will this ever return a true value which isn't literally `true`?  If not, the comparison is probably unnecessary.  And this gives me an opportunity to use the `suggest change` button!
```suggestion
    if cert_store.verify(cert_obj)
```

Reply:
I agree it is cleaner without the "== true". Adding.